### PR TITLE
Bugfix/apm 300499 event field minor bug fixes

### PR DESCRIPTION
**Pull request by kacper-olszanski**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by tomheller** [comment]
Hi @kacper-olszanski!

Could you be a bit more specific other than _Minor bug fixes_ in the pull request and also the commit message. The issue template is here to be filled out so we and others who are reading the PR get a grasp of what you are actually trying to do.

**Comment by kacper-olszanski** [comment]
> Could you be a bit more specific other than _Minor bug fixes_ in the pull request and also the commit message. The issue template is here to be filled out so we and others who are reading the PR get a grasp of what you are actually trying to do.

@tomheller 
The description of the changes is on Jira but I can copy paste it here:

fixed bugs:
- legend margin-top not working
- heatfields should be merged when start and end of heatfields are matching
- min length for fields should be set if they are too short to see

**Comment by tomheller** [comment]
@kacper-olszanski Thank you. The changes in general look good. Could you please squash your changes to get rid of the merge commit and reword the commit `fix(event-chart): Fixes various issues with the legend and heatfields.`

File: libs/barista-components/event-chart/src/merge-and-path/merge-events.ts [hunk 1]
```diff
@@ -142,16 +141,19 @@ export function dtEventChartMergeFields<T>(
...
-        currentEvent,
-        eligableEvent,
-        overlapThreshold,
-      );
+      const isOnTheSameSpotOrOverlappingALittleBit =
+        eligableEvent.x1 - currentEvent.x1 <= 3 &&
+        eligableEvent.x2 - currentEvent.x2 <= 3;
       const hasSameColor = currentEvent.color === eligableEvent.color;
-
+      console.log({
```
Comment (thomaspink) [inline libs/barista-components/event-chart/src/merge-and-path/merge-events.ts:148]: Please remove this debug log

**Review by thomaspink — CHANGES_REQUESTED** [review]
Thank you for your contribution. Before we can merge it please remove the marked console logs from the src.

**Review by thomaspink — APPROVED** [review]

**Review by ffriedl89 — APPROVED** [review]

**Review by thomaspink — APPROVED** [review]

**Comment by ffriedl89** [comment]
PR seems to have issues in the Git changeset, moved the changes over to a new branch and will merge #2132

**Pull request closed without merging**